### PR TITLE
[1.2.0] Add ability to pass get params to router

### PR DESCRIPTION
**Pull request by ngoryachev**

### CODE REVIEW

File: __tests__/router.test.js [hunk 1]
```diff
@@ -102,4 +102,14 @@ describe('Router', () => {
       wrapper.findWhere(child => child.prop('path') === Router.routes.about.path).prop('exact')
     ).toEqual(false)
   })
+
+  test('Push with location-like argument', () => {
+    Router.push(null, {
+      pathname: Router.routes.about.path,
+      search: '?the=query',
```
Comment (isnifer) [inline __tests__/router.test.js]: I think we want pass get-params as:
```js
params: {
  the: 'query',
} 
```
`params` word can be discussed

Comment (isnifer) [reply __tests__/router.test.js]: Because `tipsi-router` should not care about internal router implementation in public API

File: src/Router.js [hunk 2]
```diff
@@ -58,14 +58,25 @@ export default class TipsiRouter {
...
+    } else if (pathname) {
+      this.history[methodName](locationHolder)
+    } else {
+      throw new Error(
+        `locationHolder arg for ${methodName}(...) has neither "path" nor "pathname" key`
+      )
+    }
+  }
+
+  push(e, locationHolder, paramsOrOptions = {}) {
```
Comment (isnifer) [inline src/Router.js]: `route` was a good name for the parameter, why `locationHolder` instead?

Comment (ngoryachev) [reply src/Router.js]: "route" was too blurry definition for me. It has too many semantic clashes in tipsi-router, react-router, history so i just always asking myself "what kind of route is this route" =). I'll bring old name if you like.

Comment (isnifer) [reply src/Router.js]: I just want to remember you — if you've changed a name for a thing in `tipsi-router` you should change it on every internal router

File: src/Router.js [hunk 3]
```diff
@@ -56,16 +56,43 @@ export default class TipsiRouter {
...
 
+  getCurrentQuery() {
+    let query = this.history.location.search
+
+    if (query) {
+      const entries = query.substr(1).split('&').map(entry => entry.split('='))
+      query = entries.reduce((memo, [key, value]) => ({ ...memo, [key]: value }), {})
+    }
+
+    return query
```
Comment (isnifer) [inline src/Router.js]: Sure, you can use [qs](https://github.com/ljharb/qs)

File: src/Router.js [hunk 4]
```diff
@@ -56,16 +56,43 @@ export default class TipsiRouter {
...
-    const { config, ...params } = paramsOrOptions
-    this.history.push(path, params)
+
+    const { path, query } = route
+    const pathname = compile(path)(paramsOrOptions)
+    let search
+
+    if (query) {
+      const queryEntries = Object.entries(query).map(([key, value]) => `${key}=${value}`).join('&')
+      search = `?${queryEntries}`
```
Comment (isnifer) [inline src/Router.js]: Use [qs](https://github.com/ljharb/qs)

File: src/Router.js [hunk 5]
```diff
@@ -56,16 +57,41 @@ export default class TipsiRouter {
...
+    if (query) {
+      search = stringify(query, { addQueryPrefix: true })
+    }
+
+    const location = {
+      pathname,
+      search,
+    }
+
+    this.history[methodName](location)
```
Comment (isnifer) [inline src/Router.js]:
```js
const { path, query } = route

this.history[methodName]({
  pathname: compile(path)(paramsOrOptions),
  search: query && stringify(query, { addQueryPrefix: true }),
})
```

**Review by isnifer — APPROVED** [review]
LGTM

**Pull request merged**